### PR TITLE
backup: add scripts, general info & configuration

### PR DESCRIPTION
**Pull request by jbergstroem**
Refs: https://github.com/nodejs/build/issues/308

### CODE REVIEW

**Comment by jbergstroem** [comment]
I know this is probably out of scope for most of you but since this is critical and is currently running nightly it'd be good to have an eye or two for review/questions, @nodejs/build.

File: backup/backup_scripts/dist.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
```
Comment (thefourtheye) [inline backup/backup_scripts/dist.sh]: Why this alone is not run in bash?

Comment (jbergstroem) [inline backup/backup_scripts/dist.sh]: not really needed, but fixed anyway for consistency.

File: backup/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+| `periodic/` | periodically updated storage managed by rsnapshot |
+
+
+### Setting it up
+
+Dependencies:
+ - rsync 3.x (or newer)
+ - rsnapshot 1.2 (or newer)
+
+Note: You will likely want somewhere around ~1Tb storage.
```
Comment (thefourtheye) [inline backup/README.md]: around and `~` seem redundant here.

File: backup/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+#### Installation instructions
+
+1. Install dependencies
+2. Copy the folder `backup_scripts` to /root/backup_scripts
```
Comment (thefourtheye) [inline backup/README.md]: Copy the folder `backup_scripts` to `/root`

**Comment by thefourtheye** [comment]
Just some minor nits.

**Comment by jbergstroem** [comment]
First weekly finished; here's the current size/state:
```
70G     ./periodic/daily.0
636M    ./periodic/daily.1
715M    ./periodic/daily.2
22G     ./periodic/daily.3
102M    ./periodic/daily.4
114M    ./periodic/daily.5
2.9G    ./periodic/daily.6
181G    ./periodic/weekly.0
2.3G    ./archive/nodejs_logs
191G    ./archive/jenkins-before-job-deletion-plugin
118G    ./static/dist
6.5K    ./static/benchmark
604G    .
```

(`archive/jenkins-before-job-deletion-plugin` will be deleted shortly)

**Comment by thefourtheye** [comment]
Instead of `daily.\d` would it be better to have the date?

**Comment by jbergstroem** [comment]
@thefourtheye can't change that -- its how rsnapshot keeps track of rotation numbers (see `retain` sections in `rsnapshot.conf`).

**Comment by joaocgreis** [comment]
Rubber-stamp LGTM. The bash scripts look good, but I have no experience with rsnapshot.

What should be inside of `.jenkins_credentials`? Github token?

**Comment by mhdawson** [comment]
LGTM but again I'm not experienced with rsnapshot.  

Could you send me a copy of the benchmarking backup file, I'll load it up to validate we are capturing correctly.

**Comment by thefourtheye** [comment]
LGTM

**Pull request merged**